src/libexpr/nixexpr-as-json.cc

```diff
@@ -0,0 +1,326 @@
+#include "nixexpr.hh"
+#include "derivations.hh"
+#include "util.hh"
+
+#include <cstdlib>
+
+namespace nix {
+
+// binary operators are implemented in nixexpr.hh MakeBinOp
+
+// https://stackoverflow.com/questions/7724448
+// note: we use full jump table to make this as fast as possible
+// note: we assume valid input. errors should be handled by the nix parser
+// 93 * 7 = 651 byte
+char String_showAsJson_replace_array[93][7] = {
```

Comment:
Any reason you aren't just using something like below?
```
const char* foo[] = { "1", "2", "3", "4" };
const size_t fooLength = sizeof(foo);
```

That would remove the need to track the byte sizes require for this escaping... Alternatively just use a JSON library ;-)

Reply:
the "should be faster" reason. im not sure how your array would look in memory

Reply:
this is actually an interesting question ...

how to find the memory layout of this program?

```c
#include <stdio.h>

char array1[3][10] = { "a", "b", "findme" };

const char* array2[] = { "1", "2", "findme" };

int main() {
    printf("%s", array1[0]);
    printf("%s", array1[1]);
    printf("%s", array2[0]);
    printf("%s", array2[1]);
}
```

with `gcc a.c -O0; pahole a.out` i dont see the `findme` value

Reply:
Check this out: https://godbolt.org/z/enxGvshhY

Reply:
`const char a1[93][7]` is 5% faster than `const char* a2[]` based on [this benchmark](https://github.com/milahu/random/blob/master/cpp/benchmark-char-vs-char-pointer/bench-string-table-char-array.cpp)